packages/muze-legend/src/legend/legend-helper.js

```diff
@@ -27,16 +27,98 @@ export const getScaleInfo = (scale) => {
 };
 
 /**
- *
- *
- * @param {*} domain
- * @param {*} steps
- *
+ * @param  {} domain
+ * @param  {} domainForLegend
+ * @param  {} context
  */
-export const getInterpolatedData = (domain, steps) => {
+export const getInterpolatedArrayData = (domain, domainForLegend, context) => {
+    const { align } = context.config();
+    const availableSpace = context.measurement();
+    let currentTickValue;
+    let nextTickValue;
+    let upperBound = domainForLegend[domainForLegend.length - 1];
+    if (!Number.isInteger(domainForLegend[domainForLegend.length - 1])) {
```

Comment:
Unnecessary use of domainForLegend[domainForLegend.length - 1] 
Instead, use upperBound variable described in line 39.